activerecord/lib/active_record/associations.rb

```diff
@@ -1492,6 +1492,9 @@ def has_many(name, scope = nil, **options, &extension)
         #   When set to +true+, the association will also have its presence validated.
         #   This will validate the association itself, not the id. You can use
         #   +:inverse_of+ to avoid an extra query during validation.
+        # [:message]
+        #   Specifies the error message for +required+ option.
+        #   NOTE: If <tt>required</tt> is set to <tt>true</tt> then only this option will work.
```

Comment:
Do you mind revise this wording a bit? I don't think _only this option will work_ is the right description for this flag.

```ruby
#  NOTE: This option will be used only when `:required` flag is set to `true`.
```

Alternatively, we can also remove this note as line 1496 already made it clear that this works in conjunction with `:required`.